### ISSUE TITLE
Flutter needs a way to collect device info (e.g. chrome's about:gpu)

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#11402]
Particularly about the graphics functionality of the device in question.  This would help us not only diagnose when graphics issues are going wrong, but also collect info from users in the field.

Chrome has something along this vein called "about:gpu".

Examples of things we should dump:
- Information about how the GrContext initialized (8888? sRGB, etc)
- Make and model of the GPU
- probably lots more i'm forgetting.

We would only do this if it were useful to folks like @chinmaygarde @jason-simmons and @brianosman

**Comment by bsalomon** [comment]
How are you envisioning that this would get triggered? When an end user reports a Chrome bug we can just ask them to go to chrome:://gpu in the browser. Is it something an end user could do from any flutter app? Is it its own app?

Some things I would capture:
*Vulkan vs GL
*api version and available extensions
*Skia's computed caps including driver workarounds
*Driver version

**Comment by eseidelGoogle** [comment]
I would envision triggering something like probably from our command line tool`flutter`, possibly as part of `flutter bugreport` or maybe a separate command.  The idea is "someone has flutter misbehaving on their device, how do we tell them what 27 things we want them to collect so we can diagnose or pass along enough information to Skia to diagnose".

**Comment by brianosman** [comment]
I think this subsumes #10745.

**Comment by bsalomon** [comment]
We could write a GrContext function that dumps all the relevant info into a formatted string (json?) which could be included in the bugreport output.

**Comment by brianosman** [comment]
Yes. In at least one bug report, the user wasn't even getting a GrContext. In that case, I'm really not sure what we can do, except perhaps log more information about why it's failing. But for the general case of an app misbehaving, GrContext can dump quite a bit.

**Comment by brianosman** [comment]
I am working on a new dump() from GrContext that will provide more information, but I noticed that `flutter bugreport` (and/or) `flutter run --bugreport` are only in the 'proposal' stage? Neither one actually exists, right?

**Comment by chinmaygarde** [comment]
`bugreport` does not currently exist. But writing the contents of `GrContext::dump()` to the logs should suffice as I image bugreport details will be picked up from there.

**Comment by eseidelGoogle** [comment]
`flutter run --bug-report` exists.  @tvolkert made that.

**Comment by tvolkert** [comment]
Yep, `flutter run --bug-report` exists (follow the code from [here](https://github.com/flutter/flutter/blob/befe01989610e4e231ca1c9095fcdb7a027ede84/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L192)), so we could hook up the GrContext dump into that.

For Android, `adb bugreport` can also contain a lot of info, but there's probably value in just encapsulating it all in the `flutter` tool.

**Comment by brianosman** [comment]
I just added GrContext::dump() (https://skia.googlesource.com/skia/+/175af0d01177fc6e5a81e979cd2ae3009c375940). This returns a single JSON-formatted string with lots of information that's helpful in bug reports (GL strings for vendor, renderer, version, etc..., list of extensions, all of our derived caps information, and our shader caps information). Should I look into writing that out somewhere? I feel like any output from within Skia tends to get discarded/eaten, so I assume the logging will need to be in Flutter engine code.

**Comment by tvolkert** [comment]
We could either:

1. trigger that to be logged to the console automatically after app startup when in bug-report mode, and the contents of the logs will be captured as part of the bug report
2. expose a VM service extension to get the contents of that JSON string, then call the extension and explicitly write out the GrContext::dump() to a dedicated file when in bug-report mode.

I vote `#2`.

**Comment by eseidelGoogle** [comment]
Skia now has a tool to help with this, skqp: https://skia.org/dev/testing/skqp

**Comment by bkonyi** [comment]
According to @matanlurey, clearly we didn't need this 😄
